File: ontrack-security/src/main/java/net/nemerosa/ontrack/security/ConfidentialKey.java
package net.nemerosa.ontrack.security;

import javax.crypto.Cipher;

public interface ConfidentialKey {

    /**
     * Name of the key. This is used as the file name.
     */
    String getId();

    Cipher encrypt();

    Cipher decrypt();

    String encrypt(String plain);

    String decrypt(String crypted);
}

File: ontrack-service/src/main/java/net/nemerosa/ontrack/service/security/ConfidentialKey.kt
package net.nemerosa.ontrack.service.security

import javax.crypto.Cipher

interface ConfidentialKey {
    /**
     * Name of the key. This is used as the file name.
     */
    val id: String

    fun encrypt(): Cipher
    fun decrypt(): Cipher
    fun encrypt(plain: String): String
    fun decrypt(crypted: String): String

    fun exportKey(): String?

    fun importKey(key: String)
}